app/src/sys/unix/command.c

```diff
@@ -1,24 +1,66 @@
+#define _GNU_SOURCE
 #include "command.h"
 
+#include <errno.h>
+#include <fcntl.h>
 #include <signal.h>
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
 #include "log.h"
 
-pid_t cmd_execute(const char *path, const char *const argv[]) {
-    pid_t pid = fork();
-    if (pid == -1) {
-        perror("fork");
+int cmd_execute(const char *path, const char *const argv[], pid_t *pid) {
+    int fd[2];
+    int ret = 0;
+
+    if (0 > pipe2(fd, O_CLOEXEC)) {
+        perror("pipe2");
         return -1;
     }
-    if (pid == 0) {
+
+    *pid = fork();
+    if (*pid == -1) {
+        perror("fork");
+        ret = -1;
+        goto END;
+    }
+
+    if (*pid > 0) {
+        /* parent close write side. */
+        close(fd[1]);
+        fd[1] = -1;
+        /* and blocking read until child exec or write some fail
+         * reason in fd. */
+        if (0 > read(fd[0], &ret, sizeof(int))) {
+            perror("read");
+            ret = -1;
+            goto END;
+        }
+    } else if (*pid == 0) {
+        /* child close read side. */
+        close(fd[0]);
+        fd[0] = -1;
```

Comment:
To enable CLOEXEC without `pipe2()`, add:
```c
        if (fcntl(fd[1], F_SETFD, FD_CLOEXEC) == -1) {
            perror("fcntl");
        }
```

Reply:
will do